Marlin/src/inc/SanityCheck.h

```diff
@@ -4380,8 +4380,16 @@ static_assert(_PLUS_TEST(3), "DEFAULT_MAX_ACCELERATION values must be positive."
   #elif DISABLED(FTM_UNIFIED_BWS)
     #error "FT_MOTION requires FTM_UNIFIED_BWS to be enabled because FBS is not yet implemented."
   #endif
+  #if !HAS_X_AXIS && !HAS_Y_AXIS
+    #warning "FT_MOTION shaping needs an X or an Y axis."
+  #else
+    #define FTM_SHAPING
+  #endif
```

Comment:
Don't set conditionals and don't emit `#warning` in `SanityCheck.h`. Conditionals for FT Motion belong in `Conditionals_adv.h`. Warnings must go into `Warnings.cpp`.

Note that in the future may allow for FT Motion to be enabled with no shaping options, simply using it as an improved motion system on machines that don't need (or can't use) shaping.